client/state/products-list/selectors/is-marketplace-product.js

```diff
@@ -0,0 +1,6 @@
+import 'calypso/state/products-list/init';
+
+export function isMarketplaceProduct( state, productSlug ) {
+	const product = state.productsList?.items?.[ productSlug ];
```

Comment:
product slugs should also contain `monthly` or `yearly` for example

```
woocommerce_subscriptions_monthly: {
product_id: 903,
product_name: "WooCommerce Subscriptions",
product_slug: "woocommerce_subscriptions_monthly",
description: "",
product_type: "marketplace_plugin",
available: true,
is_domain_registration: false,
cost_display: "€19.00",
cost: 19,
currency_code: "EUR",
price_tier_list: [ ],
price_tier_usage_quantity: null,
price_tiers: [ ],
price_tier_slug: "",
},
```

how can we work around this?

Reply:
@cpapazoglou do you mean that a marketplace product should have either `yearly` or `monthly` in it's slug, or that these slugs should be persisted in state too? 

I've checked the state quickly and these seem to be persisted correctly:
![image](https://user-images.githubusercontent.com/11555574/145388088-54db6aaa-044d-4724-af04-8c85c583be9f.png)

Reply:
We've discussed that these are actually _subscription slugs_ which are different from the _product slugs_ that are used in the URLs. I'll come up with a way to translate between the two types of slugs.

Reply:
I've updated the selector to check both for dashes and underscores in the plugin slug in https://github.com/Automattic/wp-calypso/pull/58964/commits/d0fff6da04fdb33dae36aba474054dcab1a37d4f

@cpapazoglou  Can we return `true` from the selector if the slug ends in either `monthly` or `yearly`? We could also pass these from where the selector will be used.
![image](https://user-images.githubusercontent.com/11555574/145413217-7c127747-b772-4247-907e-c9c7f83de43e.png)

Reply:
@gcsecsey when checking if a `product` is a `marketplace_product` (where `product === plugin || theme || service`) we shouldn't care at all whether it has a monthly or yearly version. I would totally remove the billing period `yearly`, `monthly`, `2y` when checking the slugs.